### ISSUE TITLE
TST: (numpy dev) new failing tests

### ISSUE THREAD

**Issue by neutrinoceros** [astropy/astropy#15926]
### Description

Three tests appear to be failing in devdeps even after #15924 is dealt with. Namely
- `astropy/units/tests/test_quantity_non_ufuncs.py::TestArgFunctions::test_argsort`
- `astropy/utils/masked/tests/test_function_helpers.py::TestArgFunctions::test_argsort`
- `astropy/utils/masked/tests/test_function_helpers.py::TestSortFunctions::test_sort`

see https://github.com/astropy/astropy/actions/runs/7609867763/job/20722001589

~I note that I'm unable to reproduce these failures locally (all tests pass with numpy built from source), so either the problem is already addressed upstream and wheels are just a couple hours behind (in which case it'll resolve itself tomorrow), or this is somehow platform specific and doesn't affect macOS.~
edit: I don't know what I did wrong that day but in any case that's now incorrect, and I can now reproduce locally.

### How to Reproduce

Install numpy dev, build pyerfa + astropy against it, then run (for instance)
```shell
pytest astropy/units/tests/test_quantity_non_ufuncs.py::TestArgFunctions::test_argsort
```

### Versions

```python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import astropy; print("astropy", astropy.__version__)
import numpy; print("Numpy", numpy.__version__)
import erfa; print("pyerfa", erfa.__version__)
#import scipy; print("Scipy", scipy.__version__)
#import matplotlib; print("Matplotlib", matplotlib.__version__)
```
```
macOS-14.2.1-arm64-arm-64bit
Python 3.12.1 (main, Dec 11 2023, 18:41:50) [Clang 15.0.0 (clang-1500.0.40.1)]
astropy 6.1.dev350+g681436af1d
Numpy 2.0.0.dev0+git20240122.89eac8b
pyerfa 2.0.1.2.dev11+gc498084
```

edit: PRs opened to fix this
- #15929
- #15930 
- #15941

**Comment by pllim** [comment]
The comparison failure stems from:

https://github.com/astropy/astropy/blob/d8d595c46cda2612dbe8afca723fba46e4ccff9d/astropy/utils/masked/tests/test_function_helpers.py#L172

https://github.com/astropy/astropy/blob/d8d595c46cda2612dbe8afca723fba46e4ccff9d/astropy/utils/masked/tests/test_function_helpers.py#L159-L163

So some sort of behavior change when `MaskedNDArray` is passed into `np.argsort.` 🤷

**Comment by pllim** [comment]
The other failure is from:

https://github.com/astropy/astropy/blob/d8d595c46cda2612dbe8afca723fba46e4ccff9d/astropy/utils/masked/tests/test_function_helpers.py#L1155

Is it caused by this?

* https://github.com/numpy/numpy/pull/25437

**Comment by neutrinoceros** [comment]
It is ! good catch !
Here's a patch for this one: #15929

**Comment by pllim** [comment]
When it comes to Column, maybe @taldcroft and @mhvk can advise. 🙏

**Comment by mhvk** [comment]
We definitely want to return `np.ndarray` - is this still a problem with the revised `__array_wrap__`? Maybe I gave the wrong suggestion...

**Comment by neutrinoceros** [comment]
> We definitely want to return np.ndarray

ok thanks, I'll see what I can do about it then.

>  is this still a problem with the revised __array_wrap__?

Yes, the problem is that this test gained two different failure modes when the `__array_wrap__` numpy PR was merged. #15925 just solved the first failure, revealing the second, which is the one I described here.

**Comment by neutrinoceros** [comment]
#15929 and #15941 will fix all problems reported here. There's another, bigger issue that causes multiples failures with numpy dev but I think it deserves its own ticket, which I'll open now.
